hyperspy/io_plugins/_hierarchical.py

```diff
@@ -626,7 +634,17 @@ def overwrite_dataset(cls, group, data, key, signal_axes=None,
                     del group[key]
 
         _logger.info(f"Chunks used for saving: {chunks}")
-        cls._store_data(data, dset, group, key, chunks)
+        if data.dtype == np.dtype('O'):
+            new_data = np.empty(shape=data.shape, dtype=object)
+            shapes = np.empty(shape=data.shape, dtype=object)
+            for i in np.ndindex(data.shape):
+                new_data[i] = np.ndarray.flatten(data[i])
```

Comment:
```suggestion
                new_data[i] = data[i].flatten()
```

Reply:
The method may be marginally faster than the function but the main reason is that it is shorter to write and read!

Actually, in this case, it may be better to use `ravel` to return a view and avoid making a copy.